src/backend/functions/projects-edit.ts

```diff
@@ -0,0 +1,528 @@
+    return {
+      changeRequestId: crId,
+      implementerId,
+      wbsElementId,
+      detail: `Added ${nameOfField} "${newValue}"`
+    };
+  } else if (oldValue !== newValue) {
+    return {
+      changeRequestId: crId,
+      implementerId,
+      wbsElementId,
+      detail: `Edited ${nameOfField} from "${oldValue}" to "${newValue}"`
+    };
+  }
+  return undefined;
+};
+
+// create a change json list for a given list (rules). Only works if the elements themselves should be compared (strings)
+export const createRulesChangesJson = (
+  nameOfField: string,
+  oldArray: string[],
+  newArray: string[],
+  crId: number,
+  implementerId: number,
+  wbsElementId: number
+) => {
+  const seenOld = new Set<string>(oldArray);
+  const seenNew = new Set<string>(newArray);
+
+  const changes: { element: string; type: string }[] = [];
+
+  oldArray.forEach((element) => {
+    if (!seenNew.has(element)) {
+      changes.push({ element, type: 'Removed' });
+    }
+  });
+
+  newArray.forEach((element) => {
+    if (!seenOld.has(element)) {
+      changes.push({ element, type: 'Added new' });
+    }
+  });
+
+  return changes.map((element) => {
+    return {
+      changeRequestId: crId,
+      implementerId,
+      wbsElementId,
+      detail: `${element.type} ${nameOfField} "${element.element}"`
+    };
+  });
+};
+
+// this method creates changes for description bullet inputs
+// it returns it as an object of {deletedIds[], addedDetails[] changes[]}
+// because the deletedIds are needed for the database and the addedDetails are needed to make new ones
+export const createDescriptionBulletChangesJson = (
+  oldArray: DescriptionBullet[],
+  newArray: DescriptionBullet[],
+  crId: number,
+  implementerId: number,
+  wbsElementId: number,
+  nameOfField: string
+): {
+  deletedIds: number[];
+  addedDetails: string[];
+  editedIdsAndDetails: { id: number; detail: string }[];
+  changes: {
+    changeRequestId: number;
+    implementerId: number;
+    wbsElementId: number;
+    detail: string;
+  }[];
+} => {
+  // const seenOld = new Map<number, string>();
+  // const seenNew = new Map<number, string>();
+  // const oldArrayFiltered = oldArray.filter((element) => {
+  //   return element.dateDeleted === undefined;
+  // });
+
+  // oldArray.forEach((element) => {
+  //   seenOld.set(element.id, element.detail);
+  // });
+
+  // newArray.forEach((element) => {
+  //   seenNew.set(element.id, element.detail);
+  // });
+
+  // const changes: { element: DescriptionBullet; type: string }[] = [];
+
+  // oldArrayFiltered.forEach((element) => {
+  //   if (!seenNew.has(element.id)) {
+  //     changes.push({ element, type: 'Removed' });
+  //   }
+  // });
+
+  // newArray.forEach((element) => {
+  //   if (element.id < 0 || !seenOld.has(element.id)) {
+  //     changes.push({ element, type: 'Added new' });
+  //   } else if (seenOld.get(element.id) !== element.detail) {
+  //     changes.push({ element, type: 'Edited' });
+  //   }
+  // });
```

Comment:
is there a reason that all this commented code is still in here? if not, I'd say go ahead and take it out